docs/source/processing.rst

```diff
@@ -491,4 +491,51 @@ Both :obj:`datasets.Dataset` and :obj:`datasets.DatasetDict` objects can be save
 Controling the cache behavior
 -----------------------------------
 
-[UNDER CONSTRUCTION]
+When applying transforms on a dataset, the data are stored in cache files.
+The caching mechanism allows to reload an existing cache file if it's already been computed.
+
+Reloading a dataset is possible since the cache files are named using the dataset fingerprint, which is updated after each transform.
+
+
+Fingerprinting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The fingerprint of a dataset in a given state is an internal value computed by combining the fingerprint of the previous state and a hash of the latest transform that was applied.
+The initial fingerprint is computed using a hash of the arrow table, or a hash of the arrow files if the dataset lives on disk.
+
+For example:
+
+.. code-block::
+
+    >>> from datasets import Dataset
+    >>> dataset1 = Dataset.from_dict({"a": [0, 1, 2]})
+    >>> dataset2 = dataset1.map(lambda x: {"a": x["a"] + 1})
+    >>> print(dataset1._fingerprint, dataset2._fingerprint)
+    d19493523d95e2dc 5b86abacd4b42434
+
+The new fingerprint is a combination of the previous fingerprint and the hash of the given transform. For a transform to be hashable, it needs to be picklable using dill or pickle.
+For non-hashable transform, a random fingerprint is used and a warning is raised.
+Make sure your transforms and parameters are serializable with pickle or dill for the dataset fingerprinting and caching to work.
+If you reuse a non-hashable transform, the caching mechanism will consider it to be different from the previous calls and recompute everything.
+
+Enable or disable caching
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Locally you can prevent the library from reloading a cached file by using ``load_from_cache=False`` in transforms like :func:`datasets.Dataset.map` for example.
+You can also specify the name of path where the cache file will be written using the parameter ``cache_file_name``.
+
+It is also possible to disable caching globally with :func:`datasets.set_caching_enabled`:
+
+.. code-block::
+
+    >>> from datasets import set_caching_enabled
+    >>> set_caching_enabled(False)
+
```

Comment:
I would put the code block after the description of what disabling caching entails, people might miss some of the effects otherwise.